runelite-client/src/main/java/net/runelite/client/plugins/zalcano/ZalcanoPlugin.java

```diff
@@ -183,4 +223,47 @@ public void onProjectileMoved(ProjectileMoved event)
 			targetedGlowingRockEndCycle = projectile.getEndCycle();
 		}
 	}
+
+	@Subscribe
+	public void onHitsplatApplied(HitsplatApplied event)
+	{
+		final Actor actor = event.getActor();
+
+		if (!(actor instanceof NPC))
+		{
+			return;
+		}
+
+		int npcId = ((NPC) actor).getId();
+
+		if (!(npcId == ZALCANO_WEAKENED || npcId == ZALCANO))
+		{
+			return;
+		}
+
+		int damage = event.getHitsplat().getAmount();
+		Hitsplat.HitsplatType type = event.getHitsplat().getHitsplatType();
+
+		switch (type)
+		{
+			case DAMAGE_ME:
+				healthDamage += damage;
+				break;
+			case DAMAGE_ME_ORANGE:
+				shieldDamage += damage;
+				break;
+		}
+	}
+
+	private boolean checkArea()
+	{
+		final Player localPlayer = client.getLocalPlayer();
+		if (localPlayer == null)
+		{
+			return false;
+		}
+
+		final WorldPoint point = WorldPoint.fromLocalInstance(client, localPlayer.getLocalLocation());
```

Comment:
We think Zolcano is never an instance. Also this means you can move this off of the client thread if you need.